modules/MapGenerator/PreviewRezult.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/*************************************************************************************************
+ * Copyright 2014 JPL TSolucio, S.L. -- This file is a part of TSOLUCIO coreBOS Customizations.
+* Licensed under the vtiger CRM Public License Version 1.1 (the "License"); you may not use this
+* file except in compliance with the License. You can redistribute it and/or modify it
+* under the terms of the License. JPL TSolucio, S.L. reserves all rights not expressly
+* granted by the License. coreBOS distributed by JPL TSolucio S.L. is distributed in
+* the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. Unless required by
+* applicable law or agreed to in writing, software distributed under the License is
+* distributed on an "AS IS" BASIS, WITHOUT ANY WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific language governing
+* permissions and limitations under the License. You may obtain a copy of the License
+* at <http://corebos.org/documentation/doku.php?id=en:devel:vpl11>
+*************************************************************************************************/
+
+global $adb, $log, $current_user, $default_charset;
+
+$selectedfieldsfromhistory = $_POST['queryhistory'];
+$listoffileds = explode("FROM",$selectedfieldsfromhistory);
+$myArray=explode(",",str_replace("SELECT","",$listoffileds[0]));
```

Comment:
Dangerous str_replace:
 - you don't clean the input parameter
 - you only use uppercase, so if we get the SQL in lowercase, it will not explode or str_replace correctly
 - what happens if we have a field called "unselectfield" or something similar that contains "select"?  Get it? It is much faster and safer to just cut the first 6/7 characters from the trim of the string.